src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -0,0 +1,553 @@
+                modifiedLog |= UpdateVersionControlProvenance(runs);
+
+                modifiedLog |= UpdateRunAutomationDetails(runs);
+            }
+
+            return true;
+        }
+
+        private static bool UpdateSarifLogVersion(JObject sarifLog)
+        {
+            bool modifiedLog = false;
+
+            string version = (string)sarifLog["version"];
+            if (version != SarifUtilities.SemanticVersion)
+            {
+                sarifLog["version"] = SarifUtilities.SemanticVersion;
+                modifiedLog = true;
+            }
+
+            string schema = (string)sarifLog["$schema"];
+            if (schema != SarifUtilities.SarifSchemaUri)
+            {
+                sarifLog["$schema"] = SarifUtilities.SarifSchemaUri;
+                modifiedLog = true;
+            }
+
+            return modifiedLog;
+        }
+
+        private static bool UpdateRunAutomationDetails(JArray runs)
+        {
+            bool modifiedRun = false;
+
+            foreach (JObject run in runs)
+            {
+                // The properties comprise the run-level runAutomationId in previous versions
+                JToken logicalId = run["logicalId"];
+                JToken description = run["description"]; // 'message' object type
+                JToken instanceGuid = run["instanceGuid"];
+                JToken correlationGuid = run["correlationGuid"];
+
+                if (logicalId != null || description != null ||
+                    instanceGuid != null || correlationGuid != null)
+                {
+                    run.Remove("logicalId");
+                    run.Remove("description");
+                    run.Remove("instanceGuid");
+                    run.Remove("correlationGuid");
+
+                    var runId = new JObject();
+
+                    if (instanceGuid != null)
+                    {
+                        // We can only effectively populate the new instanceId in a case where
+                        // the log if previously uniquely identified a run by a guid.
+                        runId["instanceId"] = logicalId + "/" + instanceGuid;
+                    }
+
+                    if (description != null) { runId["description"] = description; }
+                    if (instanceGuid != null) { runId["instanceGuid"] = instanceGuid; }
+                    if (correlationGuid != null) { runId["correlationGuid"] = correlationGuid; }
+
+                    run["id"] = runId;
+
+                    modifiedRun = true;
+                }
+
+                // This property is an element of what v2 now refers to as aggregateIds
+                JToken automationLogicalId = run["automationLogicalId"];
+
+                if (automationLogicalId != null)
+                {
+                    run.Remove("automationLogicalId");
+
+                    var aggregateId = new JObject();
+
+                    // For the aggregating automat id, we have to provide only the
```

Comment:
>the [](start = 79, length = 3)

"... only the logical id component"?

Reply:
'For the aggregating automation id, we can only provide the logical component'

The reason is that we didn't previously define an instance id for this item

---
In reply to: [226764086](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226764086) [](ancestors = 226764086)